### PR TITLE
Implement minimize method, for frameless windows, for edge and gtk backend.

### PR DESCRIPTION
**Pull request by Blakeinstein**
part of #199 

This should allow the one to call webview.set_minimized(), to set the window state to minimized.

The window can then be restored by normal means.

### CODE REVIEW

**Comment by Blakeinstein** [comment]
For cocoa, 

```c
WEBVIEW_API void webview_set_minimized(webview_t w) {
  struct cocoa_webview* wv = (struct cocoa_webview*)w;
  objc_msgSend(wv->shared.window, sel_registerName("miniaturize:"), NULL);
}
```
The following leads to no effect... 
According to [documentation](https://developer.apple.com/documentation/appkit/nswindow/) it should minimize, am I doing something wrong?

**Comment by zxey** [comment]
You can check how `winit` does it [here](https://github.com/rust-windowing/winit/blob/6919c2fb2d45dc4063c9ef52a69d857ed30657c7/src/platform_impl/macos/window.rs#L667), seems like they call miniaturize in similar fashion, maybe it can be some flag we are not setting on NSWindow? There is also [`perfomMiniaturize`](https://developer.apple.com/documentation/appkit/nswindow/1419749-performminiaturize) maybe that will work?

**Comment by Blakeinstein** [comment]
performMiniaturize, and performZoom simulate minimize and maximize button press on the title bar, but that will fail if the window is rendered frameless. I will give passing the sender a shot. As for the flags I think we should be setting some flags for window will minimize in the delegate much like window will close. I will look into it.

**Comment by Blakeinstein** [comment]
Ignore that, I fixed it. Turns out I needed to add the NSWindowStyleMaskMiniaturizable flag on frameless section. Since i was using a frameless window to test. 

```cpp
if (wv->frameless) {
    style = NSWindowStyleMaskBorderless | NSWindowStyleMaskMiniaturizable;
}
```

**Comment by Blakeinstein** [comment]
The rebase was weird, but I think it worked... There shouldn't be any merge conflicts now.

File: webview-sys/webview_edge.cpp [hunk 1]
```diff
@@ -277,6 +277,36 @@ class browser_window {
                         SWP_NOZORDER | SWP_NOACTIVATE | SWP_FRAMECHANGED);
         }
     }
+    void set_minimized()
```
Comment (FrankenApps) [inline webview-sys/webview_edge.cpp:280]: This is more of a question: Might it not be more consistent to also pass a Boolean argument to this function? If it is to difficult to restore the window after minimizing, we could also require the parameter to be true for now...

**Comment by Blakeinstein** [comment]
My original line of thought was to actually pass a boolean, but I later on skipped on that since I personally felt that the user shouldnt force a restore and let the os handle restoring. This also means that should the window be unminified by the os itself, the dev wont have to check and update any flags he might be using. 

Windows has SW_Restore for unminimizing and I am sure other libs also have calls for the same. Let me know if I should Implement it.

**Comment by Blakeinstein** [comment]
I just gave this some thought and realized that instead of storing the maximized state, I should instead check window state for minimze and maximize. Should I implement both in this PR? Or should there be a seperate PR?

**Comment by zxey** [comment]
Feel free to implement them both in this PR if you want :).

**Comment by Blakeinstein** [comment]
I am sorry for the delay, college happened. If the CI succeeds, this PR is mergeable. I will create a new PR with my frameless window fixes (that work around the extra border).

**Comment by Blakeinstein** [comment]
Oh and I also added Blakeinstein/Bloop to the showcase, it would be better if we rather did the showcase additions via forms or something instead of PR, as with a pr they are marked as contributors.

**Comment by zxey** [comment]
Thank you.

**Pull request merged**